src/main/java/org/broadinstitute/consent/http/mail/SendGridAPI.java

```diff
@@ -138,14 +197,14 @@ public Optional<Response> sendDataCustodianApprovalMessage(String toAddress, Str
         return sendMessage(message);
     }
 
-    public void sendDatasetApprovedMessage(String toAddress, Writer template) throws MessagingException {
+    public Optional<Response> sendDatasetApprovedMessage(String toAddress, Writer template) throws MessagingException {
         Mail message = datasetApprovedMessage.datasetApprovedMessage(toAddress, fromAccount, template);
-        sendMessage(message);
+        return sendMessage(message);
     }
 
-    public void sendDatasetDeniedMessage(String toAddress, Writer template) throws MessagingException {
+    public Optional<Response> sendDatasetDeniedMessage(String toAddress, Writer template) throws MessagingException {
         Mail message = datasetDeniedMessage.datasetDeniedMessage(toAddress, fromAccount, template);
-        sendMessage(message);
+        return sendMessage(message);
     }
 
 }
```

Comment:
Other questions in the back of my mind that shouldn't gate this pull request:
Do we need to facilitate resending messages that fail?
Should we build a message sending queue that backs off delivery attempts when the service returns [429](https://docs.sendgrid.com/api-reference/how-to-use-the-sendgrid-v3-api/rate-limits#rate-limit-response-header)'s  
How can we leverage the SendGrid API to determine when a mailbox no longer is accepting mail and what (if anything) should we do in DUOS to deal with that.

Reply:
All good questions. I am drafting a ticket for retries, but had not considered making use of other sendgrid functionalities. I'll dig more into that too.